### PR TITLE
Account page implementation

### PR DESCRIPTION
**Pull request by kimpers**
Does not necessarily need to go into this branch, I just used it as a base.

### CODE REVIEW

File: ts/hooks/use_stake.ts [hunk 1]
```diff
@@ -89,17 +147,40 @@ export const useStake = () => {
...
+        },
+        unstake: (stakePoolData: StakePoolData[]) => {
+            unstakeFromPoolsAsync(stakePoolData).catch(handleError);
+        },
+        withdrawStake: (zrxAmount: number) => {
+            withdrawStakeAsync(zrxAmount).catch(handleError);
+        },
+        withdrawRewards: (poolIds: string[]) => {
+            withdrawRewardsAsync(poolIds).catch(handleError);
+        },
```
Comment (kimpers) [inline ts/hooks/use_stake.ts]: What do you think of exposing all these functions from the same hook?

File: ts/pages/account/dashboard.tsx [hunk 2]
```diff
@@ -193,35 +325,32 @@ export const Account: React.FC<AccountProps> = () => {
...
-
-                {_.map(MOCK_DATA.stakes, (item, index) => {
-                    return (
-                        <AccountStakeOverview
-                            key={`stake-${index}`}
-                            {...item}
+                {/* If we have data and user does not have any pools now or in next epoch show CTA */}
+                {delegatorData &&
+                    (delegatorData.forCurrentEpoch.poolData.length === 0 &&
+                    delegatorData.forNextEpoch.poolData.length === 0 ? (
```
Comment (kimpers) [inline ts/pages/account/dashboard.tsx]: **TODO:** do we show for both current and next epoch in list?

**Review by johnrjj — APPROVED** [review]

File: ts/pages/account/dashboard.tsx [hunk 3]
```diff
@@ -44,100 +53,233 @@ const MOCK_DATA = {
...
+    poolId: string;
+    rewardsInEth: BigNumber;
+}
+
+interface PoolWithStatsMap {
+    [key: string]: PoolWithStats;
+}
+
+// NOTE: TESTING ONLY! REMOVE THIS BEFORE MERGING
+const ADDRESS_OVERRIDE = '0xe36ea790bc9d7ab70c55260c66d52b1eca985f84';
```
Comment (kimpers) [inline ts/pages/account/dashboard.tsx]: **TODO:** Remove before merging

File: ts/components/definition.tsx [hunk 4]
```diff
@@ -10,6 +10,7 @@ export interface Action {
     url?: string;
     onClick?: () => void;
     shouldUseAnchorTag?: boolean;
+    isSameTarget?: boolean;
```
Comment (kimpers) [inline ts/components/definition.tsx]: **TODO:** use `onClick`  and push new route then this is not needed

**Comment by johnrjj** [comment]
Going to merge this. There's still a couple outstanding things we'll need to finish up, but I want to land this before we break for the holidays as it has a lot of code I want to touch in separate PRs.

- Loading states
- Unconnected wallet state
- 'Pending' removed stake doesn't seem to update (withdrawal works nicely, though!)

File: ts/components/definition.tsx (hunk 4 above)
Comment (johnrjj) [reply ts/components/definition.tsx]: fixed to use react-router properly :)

**Pull request merged**